### DISCUSSION TITLE
Possibly bug: Miller indices with -0

### DISCUSSION THREAD

**Discussion by GitHubUserWithoutAName** [mtex-toolbox/mtex#955] [Ask Anything] [answered]
Dear MTEX community,

I just realized that since I upgraded to MTEX 5.6.0 (from MTEX 5.4.0, I use Matlab R2019a, by the way), I get strange Miller indices such as (11-0) when plotting to a (110) pole figures, or [-0-01] /  [10-1] / [11-1] instead of [001] / [-101] / [-1-11] at the corners of the fcc fundamental region when plotting to an IPF. My code has not changed.

This looks like an inversion of Miller indices in some cases, with the very strange "-0" in some cases.

By the way: When working with fcc symmetry (e.g. Nickel), I usually define the alignment of X / Y / Z to a / b / c explicitly like this, although this should not matter for fcc: `cs = crystalSymmetry('Fm-3m', [3.52 3.52 3.52], 'X||a', 'Y||b', 'Z||c')`. Could this have to do with the strange indices? Changing the alignment of X / Y / Z to a / b / c does change the indices when visualizing the funamental region, for example.

Or is this maybe a known issues?

Thanks a lot!

**Comment by kilir** [comment, marked as answer]
Hi,
no this was a bug, you may have to use the most current version of mtex.
Cheers,
Rüdiger

**Comment by zmichels** [comment]
Hi,

I am using MTEX 5.6.1 (latest on download page), and this bug still persists.

As an example, when I run the following code, I get the attached plot in my figure window which displays [-100], [0-10], and [00-1]. Do I need to download an unreleased version for the fix?


cs = crystalSymmetry('mmm');
 
m = Miller(1,0,0,'direction',cs);
 
plot(m,'antipodal','fundamentalRegion')






Zach

**Reply by kilir** [reply]
Hi Zach,
"-1" should be fine, however the bug was related to ill places "-0".

In my case (that depends how you set the plotting convention for crystal direction plotting in mtex_settings),your code gives the following which I'd say should be fine:


<img width="235" alt="example" src="https://user-images.githubusercontent.com/11697557/114565168-5074c700-9c71-11eb-9257-8cfcb5fc68b1.png">

Rüdiger

**Reply by GitHubUserWithoutAName** [reply]
Hi Rüdiger and Zach,

for me, updating from 5.6.0 to 5.6.1 seems to have fixed the bug! Everything looks fine.

By the way, I get the same as Zach when executing the code, i.e. [-100], [0-10], and [00-1].

**Comment by zmichels** [comment]
Ok, that makes sense regarding -0. It’s not that the -1’s are incorrect, but it is awkward, and would be preferable if they were 1. I mainly have used the default settings packaged with mtex. And as I check now, I have bAxisDirection as “east”, but east in my plot is [0-10] instead of [010]. Can you suggest a combination of settings I can use that will lead to all positive 1’s in the plot?
Thanks for any help.

**Reply by kilir** [reply]
Hi Zach,
does 
```
cs = crystalSymmetry('mmm','X||a','Y||b','Z||c');
m = Miller(1,0,0,'direction',cs);
plota2north
plotb2east
plot(m,'antipodal','fundamentalRegion')
```
do it?
Cheers,
Rüüdiger

**Comment by zmichels** [comment]
Hi Rüdiger,

Thanks for helping with this.

Unfortunately, I get the same plot as before (attached) after running those lines.

Cheers,
Zach

**Reply by kilir** [reply]
Did you assign also the convention `'X||a',....`? 
Here's what I get:

<img width="222" alt="plot" src="https://user-images.githubusercontent.com/11697557/114578608-553f7800-9c7d-11eb-8fc2-54ed714d847a.png">

**Comment by zmichels** [comment]
I did, and that is not what I get.

**Comment by ralfHielscher** [comment]
The wrong vertices at the spherical sector are indeed a bug. It will be fixed in the next release.

Ralf.

**Reply by ralfHielscher** [reply]
see 066dae798bdcab6c690c73cac52b9f1ce7fca43c for the bug fix.
